Hazel/src/Platform/Windows/WindowsWindow.h

```diff
@@ -21,24 +21,35 @@ namespace Hazel {
 		inline void SetEventCallback(const EventCallbackFn& callback) override { m_Data.EventCallback = callback; }
 		void SetVSync(bool enabled) override;
 		bool IsVSync() const override;
+		void SetWindowMode(const WindowMode& mode, unsigned int width, unsigned int height) override;
 
 		inline virtual void* GetNativeWindow() const { return m_Window; }
 	private:
 		virtual void Init(const WindowProps& props);
 		virtual void Shutdown();
 	private:
 		GLFWwindow* m_Window;
+		GLFWmonitor* m_PrimaryMonitor; // Stores a reference to the primary monitor
+		GLFWvidmode m_BaseVideoMode;   // Stores the underlying video mode being used by the OS
```

Comment:
If we store the primary monitor from the first time we create the window, it's always going to choose that monitor regardless if the user changes primary monitor while the game is running. I think it's safe enough to fetch the primary monitor for each call to `SetWindowMode`.

Reply:
I think it is better to refetch those rather cache it.